book/chapters/chapter8/non-sequential_pipelines_and_tuning.qmd

```diff
@@ -561,6 +561,17 @@ Tuning the `selection` hyperparameters can help determine which of the possible
 We additionally tune the `k` hyperparameter of the KNN learner, as it may depend on the type of preprocessing performed.
 As this hyperparameter is only active when the `"classif.kknn"` path is chosen we will set a dependency (@sec-optimization-depends):
 
+```{r 05-pipelines-branch-03-prep, echo = FALSE}
+# instead of plotting, we make autoplot() save the plot so we can edit it afterwards
+# This is *not* the same as ggplot::last_plot(), but the result is easier to handle in a loop.
+plt_container = new.env()
+autoplot = function(...) {
+  # <<- doesn't seem to work on CI for some reason
+  plt_container$plt = ggplot2::autoplot(...)
+  invisible(NULL)
+}
+```
+
```

Comment:
@mb706 this does not work as intended. Please check in the pdf output but this is quite broken...

<img width="350" alt="Screenshot 2023-07-03 at 08 44 04" src="https://github.com/mlr-org/mlr3book/assets/25639974/a5ae29f7-d65e-49cc-a621-af5ab3699adf">

Also figures I previously fixed are now printing badly again, e.g. this is now a full page

<img width="589" alt="Screenshot 2023-07-03 at 08 44 31" src="https://github.com/mlr-org/mlr3book/assets/25639974/d8e02320-04df-4330-a319-6bceac478339">

Reply:
I am fixing the warning message, and apparently the autoplot() got lost when merging master. but are you sure the plot you are showing is not an artifact of how you are rendering a single file, or broken cache? This one looks fine for me

Reply:
(or maybe an old mlr3pipelines version?)